common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -549,8 +549,19 @@ def should_force_account_creation():
         return (current_provider and
                 (current_provider.skip_email_verification or current_provider.send_to_registration_first))
 
+    def is_provider_saml():
+        current_provider = provider.Registry.get_from_pipeline({'backend': current_partial.backend, 'kwargs': kwargs})
+        saml_providers_list = list(provider.Registry.get_enabled_by_backend_name('tpa-saml'))
+        return (current_provider and
+                current_provider.slug in [saml_provider.slug for saml_provider in saml_providers_list])
+
     if not user:
-        if user_exists(details or {}):
+        if is_provider_saml():
+            email = details.get('email') if details else None
+            user_details = {'email': email}
```

Comment:
these two lines could be condensed to: `user_details = {'email': details.get('email')} if details else None`

Reply:
Done